### PR TITLE
Fixes #215: rename artifact

### LINKED ISSUE #215: Rename artifact
Current artifact name is `neo4j-spark-connector`. 

We should adapt this name to match the official name of *Neo4j Connector for Apache Spark*. 
Same applies to the repo name I guess.

I propose the obvious `neo4j-connector-for-apache-spark`. I understand is quite long but I think it's more semantic.

We can get rid of the `for-` maybe... Thoughts?

### PR DESCRIPTION
**Pull request by utnaf**
Fixes #215 

Artifact is rename, and correctly generated

Screenshot 2020-11-06 at 14 22 32

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,8 +5,8 @@
     <groupId>neo4j-contrib</groupId>
     <packaging>jar</packaging>
     <version>4.0.0-alpha</version>
```
Comment (conker84) [inline pom.xml]: we also need to remove the alpha

**Review by conker84 — CHANGES_REQUESTED** [review]
I would like to have this naming convention since we need to disambiguate scala versions:

* `neo4j-connector-apache-spark-4.0.0-scala-2.11`
* `neo4j-connector-apache-spark-4.0.0-scala-2.12`

@utnaf @moxious WDYT?

**Review by conker84 — DISMISSED** [review]

**Comment by utnaf** [comment]
Screenshot 2020-11-10 at 11 16 03

**Review by conker84 — APPROVED** [review]
Thank you @utnaf !

**Pull request merged**